test/metabase/query_processor/streaming_test.clj

```diff
@@ -202,32 +239,33 @@
-                                                    (format "card/%d/query/%s" (:id card) (name export-format))
-                                                    {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}})]
-                  ((-> assertions export-format) results))
-
-                :public
-                (let [results (mt/user-http-request :rasta :get 200
-                                                    (format "public/card/%s/query/%s" public-uuid (name export-format))
-                                                    {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}})]
-                  ((-> assertions export-format) results))
-
-                :embed
-                (let [results (mt/user-http-request :rasta :get 200
-                                                    (embed-test/card-query-url card (str "/" (name export-format)))
-                                                    {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}})]
-                  ((-> assertions export-format) results))))))))))
+              (testing endpoint
+                (case endpoint
+                  :dataset
+                  (let [results (mt/user-http-request :rasta :post 200
+                                                      (format "dataset/%s" (name export-format))
+                                                      {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}}
+                                                      :query query-json
+                                                      :visualization_settings viz-settings-json)]
+                    ((-> assertions export-format) results))
+
+                  :card
+                  (let [results (mt/user-http-request :rasta :post 200
+                                                      (format "card/%d/query/%s" (:id card) (name export-format))
+                                                      {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}})]
+                    ((-> assertions export-format) results))
+
+                  :public
+                  (let [results (mt/user-http-request :rasta :get 200
+                                                      (format "public/card/%s/query/%s" public-uuid (name export-format))
+                                                      {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}})]
+                    ((-> assertions export-format) results))
+
+                  :embed
+                  (let [results (mt/user-http-request :rasta :get 200
+                                                      (embed-test/card-query-url card (str "/" (name export-format)))
+                                                      {:request-options {:as (if (= export-format :xlsx) :byte-array :string)}})]
+                    ((-> assertions export-format) results)))))))))))
```

Comment:
not specifically related but I added this `testing`context while debugging something else